doc: add a note about asset_tracker boot chain

mbolivar opened the pull request:
I personally lost some time doing some development related to this
application due to an incorrect assumption that it was using the
secure bootloader chain which is documented in ug_bootloader.rst.

In fact, by default, this application does not use the secure
bootloader chain.

In case others are confused by this, try to help by adding a short
paragraph explaining this.

tejlmand commented on applications/asset_tracker/README.rst:
```diff
@@ -155,6 +155,10 @@ For example, configure ``CONFIG_POWER_OPTIMIZATION_ENABLE`` to enable power opti
+By default, this application does **not** use the full :ref:`ug_bootloader`.
+The final merged application binary by default will boot directly into MCUboot.
+You can use the ``CONFIG_SECURE_BOOT`` option to enable the secure bootloader chain.
+
```
```suggestion
Secure Boot
========

This application supports :ref:`ug_bootloader`.
To enable Secure Boot, set ``CONFIG_SECURE_BOOT=y``.
```

mbolivar replied:
Great idea. I've pushed something halfway in between :).

tejlmand commented:
@hakonfam @oyvindronningstad is this sufficient in order to get users started, or do you think there should be more description ?
There is a link to ug_bootloader, and from there also to the Immutable bootloader sample.
So I would say this is sufficient.

hakonfam commented:
If this is not sufficient then any required changes would be in the bootloader docs. So this changeset looks good to me.

ru-fu commented on applications/asset_tracker/README.rst:
```diff
@@ -155,6 +155,12 @@ For example, configure ``CONFIG_POWER_OPTIMIZATION_ENABLE`` to enable power opti
 In |SES|, select **Project** > **Configure nRF Connect SDK project** to browse and configure these options.
 Alternatively, use the command line tool ``menuconfig`` or configure the options directly in ``prj.conf``.
 
+Secure Boot
```
Leave out the section - it does not adhere to our template. Just add the information a new paragraph.
(We're working on updating the template to add a configuration section, and will move the content into that section then.)

tejlmand replied:
But the asset tracker doc has sections with:
- User Interface
- Power Optimization

isn't the description of how to enable the immutable bootloader similar to configuring Power Optimization ?

ru-fu replied:
User interface is a section from the template, and Power Optimization shouldn't be where it currently is - it should be a subsection to Overview, and in the future it'll go into the Configuration section as well.

tejlmand replied:
> be where it currently is - it should be a subsection to Overview, and in the futu

Thanks for the clarification.

ru-fu commented on applications/asset_tracker/README.rst:
```diff
@@ -155,6 +155,12 @@ For example, configure ``CONFIG_POWER_OPTIMIZATION_ENABLE`` to enable power opti
+===========
+
+This application supports the |NCS| :ref:`ug_bootloader`, but it is disabled by default.
+To enable secure boot, set ``CONFIG_SECURE_BOOT=y``.
```
```suggestion
To enable the immutable bootloader, set ``CONFIG_SECURE_BOOT=y``.
```

NordicBuilder commented:
**Some checks failed. Please fix and resubmit.**

## Gitlint issues

Commit ea2be123b5:
1: UC2 Body does not contain a 'Signed-off-by:' line
1: UC3 Title does not follow [subsystem]: [subject] (and should not start with literal subsys:): "Update applications/asset_tracker/README.rst"

## Identity/Emails issues

ea2be123b512e9b613bb52cd2bfefd82f0a16155: author email (Marti Bolivar <marti.bolivar@nordicsemi.no>) needs to match one of the signed-off-by entries.



*Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages.*

mbolivar commented:
@ru-fu I've squashed your changes in and removed the extra section, thanks.

However, I noticed that your commit message used. 'Co-authored-by:'. It is more customary in Linux style development to use Signed-off-by: in this context as well, so I've changed to that (notice I added Torsten's SOB too). Please let me know if there's a policy where we are using our own NCS specific git log conventions for things like this instead.

ru-fu commented:
> @ru-fu I've squashed your changes in and removed the extra section, thanks.
> 
> However, I noticed that your commit message used. 'Co-authored-by:'.

I have not added any commits?!

mbolivar commented:
> I have not added any commits?!

Sorry for noise. I accepted your suggestion with the GitHub button and that created the commit. I didn't stop to think that you didn't write the commit log for that, of course!
